src/sentry/api/event_search.py

```diff
@@ -2164,6 +2164,22 @@ def result_type_fn(function_arguments, parameter_values):
             default_result_type="duration",
             redundant_grouping=True,
         ),
+        Function(
+            "var",
+            required_args=[NumericColumnNoLookup("column")],
+            aggregate=["varSamp", ArgValue("column"), None],
+            result_type_fn=reflective_result_type(),
+            default_result_type="duration",
+            redundant_grouping=True,
+        ),
+        Function(
+            "stddev",
+            required_args=[NumericColumnNoLookup("column")],
+            aggregate=["stddevSamp", ArgValue("column"), None],
```

Comment:
I had a couple options here:
- either to use varPop/stddevPop or varSamp/stddevSamp, I chose to go with the latter
- the documentation is a bit ambiguous for [varPop](https://clickhouse.tech/docs/en/sql-reference/aggregate-functions/reference/varpop/) and stddevPop, it mentions "sample" still but I think it might be calculating the variance of the population which will be more computationally intensive. Although using the population variance will let us get around the issue following
- the main reason i haven't exposed this to the UI is because when sample size is <= 1, the sample variance and standard deviation formula blows up and results in NaNs and Inf (depending on which formula you use). Using the population formula gets around that problem and returns 0
- and finally, the units for variance will be s^2, min^2 etc and we don't handle those in the UI

Reply:
@Zylphrex suggested I look into [combinators](https://clickhouse.tech/docs/en/sql-reference/aggregate-functions/combinators/), but the condition would have to have an aggregate within itself to make sure the sample size is >1...
I wanted to try something like `["ifNotFinite", [["varSamp", [ArgValue("column")]], 0], None]`
but `ifNotFinite` is not an aggregate function, so that doesn't work

Reply:
@shruthilayaj Try `varSampOrNull` as in https://github.com/getsentry/sentry/pull/24352

Reply:
OrNull doesn't work here because the varSamp and stddevSamp functions return nan/inf when the num sample = 1. OrNull works for when "If the aggregate function does not have values" according to the docs. (this is also what I noticed when I tried that combination)

Reply:
Thanks for clarifying. That makes sense.